gobblin-api/src/main/java/org/apache/gobblin/util/io/GsonInterfaceAdapter.java

```diff
@@ -189,7 +192,38 @@ private <S> S readValue(JsonObject jsonObject, TypeToken<S> defaultTypeToken) th
   }
 
   public static <T> Gson getGson(Class<T> clazz) {
-    Gson gson = new GsonBuilder().registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz)).create();
+    Gson gson = new GsonBuilder()
+        .setObjectToNumberStrategy(CustomToNumberPolicy.INTEGER_OR_LONG_OR_DOUBLE)
+        .registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz))
+        .create();
     return gson;
   }
+
+  public enum CustomToNumberPolicy implements ToNumberStrategy {
```

Comment:
nit : minor suggestion on naming :CustomToNumberPolicy sounds a bit generic, can be renamed to something like  GsonNumberParsingPolicy to reflect its purpose more carefully.

Reply:
The library implementation is 
`public enum ToNumberPolicy implements ToNumberStrategy`

Since my code is wrapper on top of this that's the reason for keeping it generic

Reply:
It can be something like LongIntNumberPolicy